src/EnergyPlus/OutsideEnergySources.cc

```diff
@@ -427,19 +496,55 @@ void OutsideEnergySourceSpecs::calculate(EnergyPlusData &state, bool runFlag, Re
+            // apply loop limits on temperature result to keep in check
+            if (this->OutletTemp < LoopMinTemp) {
+                this->OutletTemp = max(this->OutletTemp, LoopMinTemp);
+                MyLoad = this->MassFlowRate * Cp * (this->OutletTemp - this->InletTemp);
+            }
+            if (this->OutletTemp > LoopMaxTemp) {
+                this->OutletTemp = min(this->OutletTemp, LoopMaxTemp);
+                MyLoad = this->MassFlowRate * Cp * (this->OutletTemp - this->InletTemp);
+            }
+        } else if (this->EnergyType == DataPlant::PlantEquipmentType::PurchSteam) { // determine mass flow rate based on MyLoad
+            Real64 SatTemp = FluidProperties::GetSatTemperatureRefrig(state,
+                                                                      state.dataPlnt->PlantLoop(LoopNum).FluidName,
+                                                                      state.dataEnvrn->StdBaroPress,
+                                                                      state.dataPlnt->PlantLoop(LoopNum).FluidIndex,
+                                                                      RoutineName);
+
+            Real64 EnthSteamInDry = FluidProperties::GetSatEnthalpyRefrig(state,
+                                                                          state.dataPlnt->PlantLoop(this->plantLoc.loopNum).FluidName,
+                                                                          this->InletTemp,
+                                                                          1.0,
+                                                                          state.dataPlnt->PlantLoop(this->plantLoc.loopNum).FluidIndex,
+                                                                          RoutineName);
+            Real64 EnthSteamOutWet = FluidProperties::GetSatEnthalpyRefrig(state,
+                                                                           state.dataPlnt->PlantLoop(this->plantLoc.loopNum).FluidName,
+                                                                           this->InletTemp,
+                                                                           0.0,
+                                                                           state.dataPlnt->PlantLoop(this->plantLoc.loopNum).FluidIndex,
+                                                                           RoutineName);
+            Real64 LatentHeatSteam = EnthSteamInDry - EnthSteamOutWet;
+
+            Real64 SteamMdot = MyLoad / LatentHeatSteam;
+
+            PlantUtilities::SetComponentFlowRate(state, SteamMdot, this->InletNodeNum, this->OutletNodeNum, this->plantLoc);
+
+            this->MassFlowRate = SteamMdot;
+            this->OutletTemp = SatTemp;
+            this->OutletSteamQuality = 1.0;
+            MyLoad = this->MassFlowRate * LatentHeatSteam;
+            state.dataLoopNodes->Node(this->OutletNodeNum).MassFlowRate = this->MassFlowRate;
```

Comment:
also here, don't write to the node's massflow rate, just rely on SetComponentFlowRate above.